### PR TITLE
add ns-popover w/ npm auto-udpate

### LINKED ISSUE #8350: [Request] Add nsPopover
**Library name:** nsPopover
**Git repository url:** https://github.com/nohros/nsPopover
**npm package url(optional):**  
**License(s):** The MIT License (MIT)
**Official homepage:**http://www.nohros.com/engenharia/index.html
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #8350 
@pvnr0082t please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/nohros/nsPopover
 * Official website (optional, not the repository): http://nohros.com/nsPopover
 * NPM package url (optional): https://www.npmjs.com/package/ns-popover
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 107
   - Count of watchers: 12
   - Count of forks: 98
 * NPM download stats (optional):
   - Downloads in the last day: 55
   - Downloads in the last week: 315
   - Downloads in the last month: 1178

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #8350. cc @nohros

### CODE REVIEW

**Comment by pvnr0082t** [comment]
@kennynaoh Could you please confirm that if https://www.npmjs.com/package/ns-popover is officially maintained.

File: ajax/libs/nsPopover/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "version": "0.6.8",
+  "homepage": "http://nohros.com/nsPopover",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:nohros/nsPopover.git"
+  },
+  "license": "MIT",
+  "keywords": [
+    "popover",
+    "tooltip",
```
Comment (PeterDaveHello) [inline ajax/libs/nsPopover/package.json]: Any reference of "tooltip" ?

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have confirmed with author by [issue](https://github.com/nohros/nsPopover/issues/116).
And change to npm auto-update.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Comment by PeterDaveHello** [comment]
There is only one version on npm but many on GitHub, you need to manually add the other versions in this case.

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have added all the version of files.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Comment by PeterDaveHello** [comment]
@kennynaoh you should use another commit to do it or mention it in the commit message.

**Comment by kennynaoh** [comment]
first commit:
`add ns-popover@0.6.8 w/ npm auto-udpate`
`close..., cc...`
second commit:
`add ns-popover manually`
`close..., cc...`

Is this better?
Thanks!

**Comment by PeterDaveHello** [comment]
yes

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have updated the commit message.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Comment by PeterDaveHello** [comment]
The commit to add old versions should be after the commit to add this lib.

**Comment by PeterDaveHello** [comment]
@kennynaoh please also mention the reason why you manually add the old versions in the commit message.

**Comment by kennynaoh** [comment]
`add old versions of ns-popover`
`because npm auto-update can't get old versions`

Is this better?

**Comment by PeterDaveHello** [comment]
could be

**Comment by pvnr0082t** [comment]
@kennynaoh `close #8350, cc @nohros` should be in last one commit message.

**Comment by PeterDaveHello** [comment]
It's okay to have `close #8350, cc @nohros` in the first commit, but you should also mention the issue number in the second commit.

**Comment by pvnr0082t** [comment]
@kennynaoh Please mention the issue number in the second commit, thank you.

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have fixed the commit message.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Comment by PeterDaveHello** [comment]
"`because npm auto-update can't get old versions`" -> "`Manually add the other versions from git repo because npm auto-update can't get some of the old versions`" be more clear please.

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have fixed the commit message.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

**Comment by PeterDaveHello** [comment]
This project seems to need grunt build to have the distribution files, I wonder if we could directly use the source file, @kennynaoh have you confirmed with the author?

**Comment by kennynaoh** [comment]
@pvnr0082t 
I have confirm with author.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
@kennynaoh @pvnr0082t multi-commit PR need multi reviewers.

**Comment by x09326** [comment]
LGTM

**Pull request merged**